src/custom/utils/operator/errors/QuoteError.ts

```diff
@@ -0,0 +1,84 @@
+import { ApiErrorCodes } from './OperatorError'
+
+export interface QuoteErrorObject {
+  errorType: QuoteErrorCodes
+  description: string
+}
+
+// Conforms to backend API
+// https://github.com/gnosis/gp-v2-services/blob/0bd5f7743bebaa5acd3be13e35ede2326a096f14/orderbook/openapi.yml#L562
+export enum QuoteErrorCodes {
+  InsufficientLiquidity = 'InsufficientLiquidity',
+  FeeExceedsFrom = 'FeeExceedsFrom',
+  UNHANDLED_ERROR = 'UNHANDLED_ERROR'
+}
+
+export enum QuoteErrorDetails {
+  InsufficientLiquidity = 'Token pair selected has insufficient liquidity',
+  FeeExceedsFrom = 'Current fee exceeds input "from" amount',
+  UNHANDLED_ERROR = 'An error occurred while fetching the quote information'
+}
+
+export function mapOperatorErrorToQuoteError(errorType?: ApiErrorCodes): QuoteErrorObject {
+  switch (errorType) {
+    case ApiErrorCodes.NotFound:
+      return { errorType: QuoteErrorCodes.InsufficientLiquidity, description: QuoteErrorDetails.InsufficientLiquidity }
+    default:
+      return { errorType: QuoteErrorCodes.UNHANDLED_ERROR, description: QuoteErrorDetails.UNHANDLED_ERROR }
+  }
+}
+
+export default class QuoteError extends Error {
+  name = 'QuoteErrorObject'
+  type: QuoteErrorCodes
+  description: QuoteErrorObject['description']
+
+  // Status 400 errors
+  // https://github.com/gnosis/gp-v2-services/blob/9014ae55412a356e46343e051aefeb683cc69c41/orderbook/openapi.yml#L563
+  static quoteErrorDetails = QuoteErrorDetails
+
+  private static async _getErrorMessage(response: Response) {
+    try {
+      const orderPostError: QuoteErrorObject = await response.json()
+
+      if (orderPostError.errorType) {
+        const errorMessage = QuoteError.quoteErrorDetails[orderPostError.errorType]
+        // shouldn't fall through as this error constructor expects the error code to exist but just in case
+        return errorMessage || 'Error type exists but no valid error details found.'
```

Comment:
Same as the other comment. If this will be displayed to the user, I rather have the original error name.

Reply:
agreed, prev comment